### ISSUE TITLE
[pkg/stanza] Container parser not playing well with downstream operator that experiences an error

### ISSUE THREAD

**Issue by TylerHelmuth** [open-telemetry/opentelemetry-collector-contrib#34295]
### Component(s)

pkg/stanza

## Description
I'm seeing strange behavior when I use the container parser chained with a regex parser that sometimes fails (because the regex is not always present in the log). When the `regex_parser` fails, logs collection stops.  I believe this is because the container parser currently breaks out of the `crioConsumer` goroutine if the `Write` returns an error:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/d8085724ce153870078e21004f8e68d25175768e/pkg/stanza/operator/parser/container/parser.go#L156-L169

## Steps to Reproduce

Use a filelogreceiver config such as 

```yaml
filelog:
        exclude: []
        include:
        - /var/log/pods/*/*/*.log
        include_file_name: false
        include_file_path: true
        operators:
        - id: container-parser
          max_log_size: 102400
          type: container
        - type: regex_parser
          regexp: '\slevel=(?P<severity_field>\w+)\s'
          on_error: send_quiet
          severity:
            parse_from: attributes.severity_field
        retry_on_failure:
          enabled: true
        start_at: end
```

Send through a log that does not match the regex parser and then send through a log that does match.

## Expected Result

Both logs are sent, with the second having a severity set.

## Actual Result

Neither logs are sent.

### Collector version

v0.105.0

**Comment by TylerHelmuth** [comment]
Is the solution to remove that `return` statement?

**Comment by ChrsMark** [comment]
Yeap, I just verified that. After the first error the "failed" log is exported but then because of the return no other logs are consumed. That's definitely problematic. Removing the mentioned `return` statement sounds correct and seems to solve the issue.

This `return` was recently [added](https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/d31bc2e1cd6c2f7b8b06e4219d191c7fda81527c#diff-3f65701685094531f52b03700a138417be4cb0038f6acf7fea5896a791cf7f9bR165) and it seems a bit weird to have the processing/parsing to stop/hang if one downstream has "failed". That doesn't look like the intended behavior for a filelog's pipeline.  

This makes me think of https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/d31bc2e1cd6c2f7b8b06e4219d191c7fda81527c#diff-1818c0cdb6108c178ca89cf5ee768fba0a0de451e9b6431464429d6bf0d0a823R60 as well. @djaglowski do you think it's correct to return immediately if a downstream operator has failed 🤔 ? If the purpose of https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33847 was to ensure that errors are logged (based on https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33783), then we should only log and continue without returning.

**Comment by djaglowski** [comment]
@ChrsMark I agree with your assessment. I think the loops were overlooked in both cases.

**Comment by ChrsMark** [comment]
I can prepare a patch for these and look around for any other place that might hit something similar. 
I will try to cover these with tests as well.

**Comment by gbxavier** [comment]
I observed the same behavior using the container parser followed by a JSON parser. The following log line makes the collector stop consuming logs after it fails to parse the logline as JSON.
```
2024-05-17T09:29:45.145553464Z stderr F ts=2024-05-17T09:29:45.145Z caller=node_exporter.go:193 level=info msg="Starting node_exporter" version="(version=1.8.0, branch=HEAD, revision=cadb1d1190ad95c66b951758f01ff4c94e55e6ce)"
```

Thanks a lot for reporting and fixing this! Would you happen to have a timeline for when it will be released (and available for usage)?

**Comment by ChrsMark** [comment]
Hey @gbxavier , based on the [Release Schedule](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/release.md#release-schedule) this should be available at `v0.107.0` (around _2024-08-12_).

**Comment by ChrsMark** [comment]
@TylerHelmuth @djaglowski shall we close this one or we have anything else still pending?
